code/backend/batch/utilities/helpers/embedders/PushEmbedder.py

```diff
@@ -0,0 +1,55 @@
+import logging
+from typing import List
+
+from ..AzureBlobStorageClient import AzureBlobStorageClient
+
+from ..config.EmbeddingConfig import EmbeddingConfig
+from ..config.ConfigHelper import ConfigHelper
+
+from .EmbedderBase import EmbedderBase
+from ..AzureSearchHelper import AzureSearchHelper
+from ..DocumentLoadingHelper import DocumentLoading
+from ..DocumentChunkingHelper import DocumentChunking
+from ...common.SourceDocument import SourceDocument
+
+logger = logging.getLogger(__name__)
+
+
+class PushEmbedder(EmbedderBase):
+    def __init__(self, blob_client: AzureBlobStorageClient):
+        self.blob_client = blob_client
+        config = ConfigHelper.get_active_config_or_default()
+        self.processor_map = {}
+        for processor in config.document_processors:
+            ext = processor.document_type.lower()
+            self.processor_map[ext] = processor
+
+    def embed_file(self, source_url: str, file_name: str):
+        file_extension = file_name.split(".")[-1]
+        processor = self.processor_map.get(file_extension)
+        self.__embed(source_url=source_url, processor=processor)
+        if file_extension != "url":
```

Comment:
Instead of checking for `"url"` here, which seems somewhat arbitrary and hardcoded, what do you think about adding a third, optional parameter `metadata`, and then passing in `{"embeddings_added": "true"}` where required?

Reply:
I agree but `embed_file` is an abstract method with 2 parameters being passed. `file_name` is not being used in IV Embedder. If we add another parameter only for Push, I think it starts loosing abstract method concept. What do you think?

Reply:
OK nw